### ISSUE TITLE
[3] Online Decision Transformer

### ISSUE THREAD

**Issue by ejjun92** [ejjun92/Autonomous-Driving-Papers#3]
최근 연구에서 offline RL을 sequence modeling 문제로 정의함.
그러나, 실용적인 RL 구현은 online component 포함.
즉, passive offline 데이터셋에 pretrain 된 policy를 task-specific하게 환경과 interaction 하면서 finetuning함.

Online Decision Transformer (ODT) 제안함.
sequence modeling RL인 decision transformer 기반
통합 프레임워크로 offline pretraining + online finetuning 수행.
online 상호작용의 비용이 많이드는 시나리오에 적합.

sequence-level entropy regularizers + autoregressive modeling objective

1. 온라인 단계에서 exploration objective: deterministic policies -> stochastic policies 전환
2. ODT의 architecture와 training protocol와 일관성을 유지하는새로운 replay buffer 개발
buffer에서 trajectory 저장 
online rollout을 통해 채워짐
return-conditioned policies를 parameterize 하므로
online rollout 중 원하는 return을 지정하는 전략 -> hindsight experience replay

#### offline RL
offline 데이터셋으로 agent가 trajectory의 likelihood가 autoregressively maximize 하도록 학습됨
이러한 패러다임은 
offline RL을 supervised learning 문제로 바꿈

supervised learning 패러다임이 온라인 환경에 확장될 수 있는지는 open question...

#### exploration

offline RL에 대한 전통적인 supervised learning objective인 MSE는 online 환경에서 충분 X

standard online 알고리즘의 경우, 
offline 데이터에 대한 접근이 online 성능에 영향을 미치지 않거나 오히려 부정적인 영향을 줄 수 있음.

offline pretraining 후 online finetuning에 대한 전체 파이프라인은 
training objective와 protocol에 대한 신중한 고려가 필요

#### Transformers for RL

RL을 supervised learning paradigm으로 build.
Q-function이나 policy gradient 학습하는 대신,
task specification (target goal or return)에 조건이 있는 action sequence의 예측 모델링에 초점.

[Chen et al., 2021]: model-free context-conditioned policy로 transformer 학습
[Janner et al., 2021]: policy와 model로 transformer 학습

#### Offline RL + Online Finetuning

offline 데이터셋에 pre-training / online 환경에 fine-tuning

[Nair et al., 2020]: offline or off-policy RL을 offline pre-training & online finetuning에 단순 적용하는 것은 오히려 성능 악화 초래할 수 있음.
off-policy 방법의 성능 저하는 off-policy bootstrapping으로 설명할 수 있음.
offline RL의 online finetuning에서의 성능 저하는 excess conservatism으로 설명할 수 있음. 이는 offline 환경에서 out-of-distribution 상태의 value overestimation을 방지하기 위해 필요함. 
-> AWAC: offline & oneline training에 모두 잘 작동하는 알고리즘 제안

[Kostrikov et al., 2021]: expectile-based implicit Q-learning 제안.
out-of-distribution action을 피하기 위해, behavior cloning을 통해 policy를 추출함으로써,
offline RL의 online finetuning 성능 향상시킴

[Lee et al., 2021] offline training 중 conservatism 유지하기 위해 
balanced replay 방식 + Q function의 ensemble을 통해 offline-online 설정에 대응

[Lu et al., 2021] AWAC 개선.
online finetuning 단계에서 collapse 해결하기 위해,
positive sampling과 exploration 통합

-> positive sampling과 exploration이 good online finetuning에 중요하다!!!
